pkg/scalers/stan_scaler.go

```diff
@@ -108,30 +108,51 @@ func parseStanMetadata(metadata map[string]string) (stanMetadata, error) {
 
 // IsActive determines if we need to scale from zero
 func (s *stanScaler) IsActive(ctx context.Context) (bool, error) {
-	resp, err := http.Get(s.getMonitoringEndpoint())
+	monitoringEndpoint := s.getMonitoringEndpoint()
+
+	resp, err := http.Get(monitoringEndpoint)
 	if err != nil {
 		stanLog.Error(err, "Unable to access the nats streaming broker monitoring endpoint", "natsServerMonitoringEndpoint", s.metadata.natsServerMonitoringEndpoint)
 		return false, err
 	}
+
+	if resp.StatusCode == 404 {
+		baseResp, _ := http.Get(s.getSTANChannelsEndpoint())
+
+		if baseResp.StatusCode == 404 {
+			stanLog.Info("Streaming broker endpoint returned 404. Please ensure it has been created", "url", monitoringEndpoint, "channelName", s.metadata.subject)
+
+		} else {
+			stanLog.Info("Unable to connect to STAN. Please ensure you have configured the ScaledObject with the correct endpoint.", "baseResp.StatusCode", baseResp.StatusCode, "natsServerMonitoringEndpoint", s.metadata.natsServerMonitoringEndpoint)
+		}
+
+		return false, err
+	}
+
 	defer resp.Body.Close()
 	json.NewDecoder(resp.Body).Decode(&s.channelInfo)
 
 	return s.hasPendingMessage() || s.getMaxMsgLag() > 0, nil
 }
 
+func (s *stanScaler) getSTANChannelsEndpoint() string {
+	return "http://" + s.metadata.natsServerMonitoringEndpoint + "/streaming/channelsz"
+}
+
 func (s *stanScaler) getMonitoringEndpoint() string {
-	return "http://" + s.metadata.natsServerMonitoringEndpoint + "/streaming/channelsz?" + "channel=" + s.metadata.subject + "&subs=1"
+	return s.getSTANChannelsEndpoint() + "?channel=" + s.metadata.subject + "&subs=1"
 }
 
 func (s *stanScaler) getTotalMessages() int64 {
-	return s.channelInfo.MsgCount
+	return s.channelInfo.LastSequence
```

Comment:
Looking at the code, I don't think this method is actually used anywhere, right? 

I think `getMaxMsgLag()` below calls `return s.channelInfo.MsgCount - maxValue` and that's probably the part @balchua meant needs to be changed if I'm not mistaken. 

Sorry if I misunderstood the issue. I'm trying to catch up on few things.

Reply:
You are right. I updated the incorrect line. `getTotalMessages` is never called so I removed it this time.